lib/vsc/administration/ldapsync.py

```diff
@@ -121,7 +121,16 @@ def sync_altered_accounts(self, last, dry_run=True):
                 logging.warning("Converting unicode to ascii for gecos resulting in %s", gecos)
             logging.debug('fetching public key')
 
-            public_keys = [str(x.pubkey) for x in self.client.get_public_keys(account.vsc_id)]
+            public_keys = []
+            for something in self.client.get_public_keys(account.vsc_id):
+                logging.debug("Got pubkey %s for account %s", something.pubkey, account.vsc_id)
+                try:
+                    pk = str(something.pubkey)
+                except UnicodeEncodeError:
+                    pk = something.pubkey.encode('ascii', 'ignore')
```

Comment:
add a test with a unicode error key?

Reply:
Well, not sure how to test since it is the actual adding in the LDAP that fails. But we need to mock that anyway. So I am not sure what adding some unicode test would add besides verifying that we get ascii.